docs/overview/building-surveys.md

```diff
@@ -0,0 +1,341 @@
+
+```json
+'{
+  "elements": [{
+    "name": "q1",
+    "title": "Question 1",
+    "type": "text",
+    "inputType": "range",
+    "min": 0,
+    "max": 100,
+    "step": 10,
+    "defaultValue": 50,
+    "isRequired": true
+  }, {
+    "name": "q2",
+    "title": "Question 2",
+    "type": "text",
+    "inputType": "range",
+    "min": 0,
+    "max": 100,
+    "step": 10,
+    "defaultValue": 50,
+    "isRequired": true
+  }, 
+  // ... more questions ...
+  {
+    "name": "qN",
+    "title": "Question N",
+    "type": "text",
+    "inputType": "range",
+    "min": 0,
+    "max": 100,
+    "step": 10,
+    "defaultValue": 50,
+    "isRequired": true
+  }]
+}'
+```
+
+But another option is to use the `survey_function` to loop over your question prompts and dynamically add questions to your survey. This way you can separate the information that changes across questions from the things that are common across all questions, which can make it easier to make changes and prevent errors (compared to working with very large and repetitive JSON structures).
+
+```javascript
+const survey_function = (survey) => {
+
+  // this array stores any information that changes across questions
+  const questions = [
+    {title: "Question 1", name: "q1"},
+    {title: "Question 2", name: "q2"},
+    // ... more question variables ...
+    {title: "Question N", name: "qN"}
+  ];
+
+  const page = survey.addNewPage("questions");
+
+  for (let i=0; i<questions.length; i++) {
+    let q = page.addNewQuestion("text", questions[i].name, i);
+    q.title = questions[i].title;
+    q.inputType = "range";
+    q.min = 0;
+    q.max = 100;
+    q.step = 10;
+    q.defaultValue = 50;
+    q.isRequired = true;
+  }
+
+}
+```
+
+!!! tip "Use the matrix question type for Likert-style tables"
+    The example above was used to illustrate how you can loop over information to programmatically construct a series of questions. But in cases where you have different question prompts that all use the same multiple choice options, you might prefer to use the the SurveyJS ["matrix" question type](https://surveyjs.io/form-library/examples/single-selection-matrix-table-question/jquery). This question type creates a table where each row is a question and each column is a response option. The table format is often used for Likert scales, satisfaction surveys, etc. You can even create a table that repeats a set of questions with different response types using the SurveyJS [multi-select matrix](https://surveyjs.io/form-library/examples/multi-select-matrix-question/jquery#) ("matrixdropdown") question type.
+
+
+Another option is to create your JSON configuration programmatically. This can be done in JavaScript that runs when the experiment first loads (before `jsPsych.run` is called). You can find an example that demonstrates this approach in the `survey` plugin's [Examples](../plugins/survey.md#examples) section.
```

Comment:
Do you think it's worth having a section of this about controlling the appearance of the SurveyJS content via CSS?

Reply:
Yes! Can I add this later?

Reply:
Yeah of course!